### DISCUSSION TITLE
Set creation workflow questions

### DISCUSSION THREAD

**Discussion by scott-wilson** [Autodesk/maya-usd#1996] [Q&A]
Hey all,

My studio is currently investigating using USD for creating sets. I have a few questions about how to best do it using the stock Maya USD plugin (currently testing 0.13). But first, here's a quick overview of how we expect USD sets would look in the pipeline.

# Stage 1 - Publishing Props/Set Elements

Let's say we have the following set elements/props. The bedroom (floor, walls, ceiling - static), bed (deforming), chair (static), table (static), and curtain (deforming). We publish out a Maya scene + USD scene (neither depend on the other), and move onto the next stage.

# Stage 2 - Creating a Set

At this point, we want to bring everything together in a set file. In this example, we could grab the published USDs of the bedroom, chair, and table and place it directly in a USD layer. But, the curtains and bed should(?) currently stay out of the USD layer, since the USD referencing from Maya isn't 100% implemented yet (works when importing a USD, but not when doing create -> USD layer from file). I'm guessing we'd have a Maya scene that would contain both the USD layer + deforming assets. This gets published out as both a Maya scene and a USD scene (the Maya scene depends on the USD scene).

# Stage 3 - Layout/Animation

Layout references or imports the Maya scene with the USD data. If they want to modify the USD data, then that'll have to be in its own layer. (I guess we wouldn't be able to reference in the bedroom in that case?)

They do the work they need to do, then publish out the caches including the changes to the USD data (as a layer).

Animation will pull in the USD for that layout set + rigs and etc for anything deforming, and do what it needs to do.

# Lighting

At this point, there should be USD data for everything, so all that is left is to merge in any other departments (such as simulation), shaders, etc into the DCC, then light and render.

# The Questions

- What's the current recommended workflow for assets that needs to contain child assets with Maya data? For example, a bedroom with a curtain that needs to deform inside of a house.
- Currently (as of 0.13) file -> export USD does not include the child USD file with the export. I'd love to be able to have an artist bring in an asset, animate it as needed, then when we hit export, a USD scene is created with the cached Maya data, and also a pointer to the child USD assets so I could load that USD and get what came from animation.
- The root nodes of our assets contain some extra info that we use to link the asset to our database. What's the best way to represent those nodes in USD + have those attributes saved in the USD file (bonus if we can access that in Maya)? I checked the discussion board, and it looks like there isn't support for arbitrary data from USD -> Maya support yet.

Thank you for your time!

**Comment by womanyoyoyo-adsk** [comment]
Hi @scott-wilson, sorry it took a while to get back to you. Currently we are supporting these workflows for layout and animation:

Exporting and Referencing USD files: https://www.youtube.com/watch?v=-vs5YFxWTdw&t=1s

Caching Maya Animations to USD: [https://www.youtube.com/watch?v=XW9Pu2ep-kM](https://www.youtube.com/watch?v=XW9Pu2ep-kM)

I haven't answered all your questions but hopefully this helps for now.
